src/main/java/uk/gov/companieshouse/company/links/processor/ChargesStreamProcessor.java

```diff
@@ -40,6 +41,54 @@ public ChargesStreamProcessor(CompanyProfileService companyProfileService,
         this.logger = logger;
     }
 
+    /**
+     * Process a ResourceChangedData message for delete.
+     */
+    public void processDelete(Message<ResourceChangedData> resourceChangedMessage)
+        throws JsonProcessingException {
+        MessageHeaders headers = resourceChangedMessage.getHeaders();
+        final ResourceChangedData payload = resourceChangedMessage.getPayload();
+        final String logContext = payload.getContextId();
+        final Map<String, Object> logMap = new HashMap<>();
+
+        // the resource_id field returned represents the charges record's company number
+        final String companyNumber = extractCompanyNumber(payload.getResourceUri());
+        if (StringUtils.isEmpty(companyNumber)) {
+            logger.error("Company number is empty or null");
+            throw new NonRetryableErrorException("Company number is empty or null");
+        }
+
+        logger.trace(String.format("Resource changed message for delete event of kind %s "
+            + "for company number %s retrieved", payload.getResourceKind(), companyNumber));
+
+        final ApiResponse<CompanyProfile> response =
+            getCompanyProfileApi(logContext, logMap, companyNumber);
+
+        var data = response.getData().getData();
+        var links = data.getLinks();
+
+        if (links != null && links.getCharges() == null) {
```

Comment:
Hopefully you will remove this logic in next story as delete removes link instead of adding